### ISSUE TITLE
unmask_passticket.py

### ISSUE THREAD

**Issue by HeadleyThompsett** [bigendiansmalls/passticket-tools#2]
Thanks for this, but it seems overly complex.
How about

#!/bin/rexx               
Arg mask .
key  = "BBFCD18E826660B6"x
Say C2x(BitXOR(X2c(mask),key)) 
Return

**Comment by bigendiansmalls** [comment]
Yes it was :) Haven’t gotten around to updating. If you’d like to submit a PR, will test and pull.

On Apr 6, 2021, at 16:23, HeadleyThompsett ***@***.***> wrote:

﻿

Thanks for this, but it seems overly complex.
How about

#!/bin/rexx
Arg mask .
key = "BBFCD18E826660B6"x
Say C2x(BitXOR(X2c(mask),key))
Return

**Comment by bigendiansmalls** [comment]
Although we do this offline, not on the system - so need python vs rexx

On Apr 6, 2021, at 16:23, HeadleyThompsett ***@***.***> wrote:

﻿

Thanks for this, but it seems overly complex.
How about

#!/bin/rexx
Arg mask .
key = "BBFCD18E826660B6"x
Say C2x(BitXOR(X2c(mask),key))
Return

**Comment by HeadleyThompsett** [comment]
I wish I knew enough about python to code this in python, but since I have rexx installed on my linux & windows machines & don't have python on any of my mainframes, this seemed better.

But here is similar in bash..

#!/bin/bash
mask=0x${1:-"0123456789abcdef"}
key=0xBBFCD18E826660B6 
printf "%X\n" "$(( $mask ^ key ))"

**Comment by bigendiansmalls** [comment]
If it’s just an xor with a static key, I can write it and test it out.

On Apr 6, 2021, at 20:54, HeadleyThompsett ***@***.***> wrote:

﻿

I wish I knew enough about python to code this in python, but since I have rexx installed on my linux & windows machines & don't have python on any of my mainframes, this seemed better.

But here is similar in bash..

#!/bin/bash
mask=0x${1:-"0123456789abcdef"}
key=0xBBFCD18E826660B6
printf "%X\n" "$(( $mask ^ key ))"
